pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -998,37 +998,70 @@ public ResultSet getTables(String catalog,
     fields[9] = new Field("REF_GENERATION", Oid.VARCHAR);
 
     String schemaName = getCrateSchemaName();
-    String stmt = "select " + schemaName + ", table_name" +
-      " from information_schema.tables" +
-      createInfoSchemaTableWhereClause(schemaName, schemaPattern, tableNamePattern, null) +
-      " order by " + schemaName + ", table_name";
+    String infoSchemaTableWhereClause = createInfoSchemaTableWhereClause(schemaName, schemaPattern, tableNamePattern,
+            null).toString();
+    String stmt = getTablesStatement(schemaName, infoSchemaTableWhereClause);
+
     ResultSet rs = connection.createStatement().executeQuery(stmt);
 
     List<byte[][]> tuples = new ArrayList<>();
     while (rs.next()) {
       byte[][] tuple = new byte[fields.length][];
 
       String schema = rs.getString(schemaName);
-      if ("sys".equals(schema) || "information_schema".equals(schema)) {
-        tuple[3] = connection.encodeString("SYSTEM TABLE");
-      } else {
-        tuple[3] = connection.encodeString("TABLE");
-      }
 
-      tuple[0] = null;
-      tuple[2] = rs.getBytes("table_name");
       tuple[1] = schema.getBytes();
+      tuple[2] = rs.getBytes("table_name");
       tuple[4] = connection.encodeString("");
       tuple[5] = null;
       tuple[6] = null;
       tuple[7] = null;
-      tuple[8] = connection.encodeString("_id");
-      tuple[9] = connection.encodeString("SYSTEM");
+
+      if (getCrateVersion().before("2.0.0")) {
+        tuple[0] = null;
+        if ("sys".equals(schema) || "information_schema".equals(schema)) {
+          tuple[3] = connection.encodeString("SYSTEM TABLE");
+        } else {
+          tuple[3] = connection.encodeString("TABLE");
+        }
+        tuple[8] = connection.encodeString("_id");
+        tuple[9] = connection.encodeString("SYSTEM");
+      } else {
+        tuple[0] = rs.getBytes("table_catalog");
+        tuple[3] = rs.getBytes("table_type");
+        tuple[8] = rs.getBytes("self_referencing_column_name");
+        tuple[9] = connection.encodeString(getReferenceGeneration(rs.getString("reference_generation")));
+      }
+
       tuples.add(tuple);
     }
     return ((BaseStatement) createMetaDataStatement()).createDriverResultSet(fields, tuples);
   }
 
+  private String getTablesStatement(String schemaName, String infoSchemaTableWhereClause) throws SQLException {
+    String select;
+    if (getCrateVersion().before("2.0.0")) {
+      select = "select " + schemaName + ", table_name";
+    } else {
+      select = "select " + schemaName + ", table_name, table_catalog, table_type, self_referencing_column_name, reference_generation";
+    }
+    return select +
+            " from information_schema.tables" + infoSchemaTableWhereClause +
+            " order by " + schemaName + ", table_name";
+  }
+
+  private String getReferenceGeneration(String referenceGeneration) {
+    switch (referenceGeneration) {
```

Comment:
the Switch statement uses .equals() with strings - would it be better to write this as a chain of ifs?